### PR TITLE
fixed: export LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by akva2**
certain modules (ert, opm-parser) requires shared libraries due
to python bindings. this causes havoc since we install to non-standard
directories during CI. fix this by setting the appropriate
LD_LIBRARY_PATH during the ci job

### CODE REVIEW

**Comment by bska** [comment]
Isn't RPATH supposed to solve this?

**Comment by akva2** [comment]
lol. for the umpteenth time; yes it is supposed to but it does *NOT* solve it.

**Comment by bska** [comment]
> yes it is supposed to but it does NOT solve it.

I see.  Do we know if the symptom presents in all cases, on Jenkins only, on some other subgroup of systems?

**Comment by akva2** [comment]
i have never seen it outside jenkins. it just showed up now, for the first time for the compareECL binary. it has been biting use in relation to summary.x for a long time. just to prove the instability: even though they run under what is supposed to be 100% identical circumstances, we have been getting norne graphs from the pr builder run trigger, but not the post-commit builder. i have absolutely no idea what differs between those (they run 100% the same scripts, with the same toolchains and whatnot).

**Comment by akva2** [comment]
tldr; i'm sure there's a perfectly reasonable explanation for this, i just don't want to spend time hunting it

**Comment by jokva** [comment]
Why don't we just build statically on jenkins, then? Seems like it would solve this issue.

**Comment by akva2** [comment]
because python stuff needs shared libraries.

**Comment by jokva** [comment]
We don't have to build that on jenkins, do we?

**Comment by akva2** [comment]
we do. there are tests depending on it.

**Comment by jokva** [comment]
Weren't they just killed off in favour of the C++ binaries?

**Comment by akva2** [comment]
nobody has done any killing of anything.

**Comment by pgdr** [comment]
@akva2 why wasn't that handled when the c++ test framework was merged?  I thought that was the sole purpose of the work done there—to remove python testing?

**Comment by akva2** [comment]
because i have never been tasked with removing anything, i've been tasked with adding.

**Comment by pgdr** [comment]
@akva2 then I simply misunderstood; I thought the task was _replacing_ ... my bad!

Anyway, it shouldn't be much work (as in, a simple flag to cmake) to build ert and opm-parser static.

**Comment by akva2** [comment]
yes. building static is simple. but i won't remove that python code, i won't touch it with a 100 foot pole. you want to remove it go right ahead, but you have to  handle your internal politics yourself.  i'll stay right out of it far over here.

**Comment by bska** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
fixed through static libraries.

**Comment by akva2** [comment]
and just a fyi: i finally found the root cause of the problem. there was a post-commit builder for ERT on jenkins prior to my taking over that i hadn't looked at. it was installing things on the system. so even though i "knew" i had nuked the binaries in /usr/local/bin and thus thought i had taken care of the problem, they were back. and these binaries were looking for a shared ert that was installed to /usr/local without running ldconfig, thus the linker couldn't find them.

**Comment by bska** [comment]
> there was a post-commit builder for ERT on jenkins [that] was installing things on the system.

Glad you figured that out.  Great work!
